src/client/pythonEnvironments/discovery/locators/services/pipEnvHelper.ts

```diff
@@ -0,0 +1,127 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+import * as path from 'path';
+import { getEnvironmentVariable } from '../../../../common/utils/platform';
+import { pathExists, readFile } from '../../../common/externalDependencies';
+
+/**
+ * Returns the path to Pipfile associated with the provided directory.
+ * @param searchDir the directory to look into
+ * @param lookIntoParentDirectories set to true if we should also search for Pipfile in parent directory
+ */
+async function getAssociatedPipfile(
+    searchDir: string,
+    lookIntoParentDirectories: boolean,
+): Promise<string | undefined> {
+    const pipFileName = getEnvironmentVariable('PIPENV_PIPFILE') || 'Pipfile';
+    let depthToSearch = 1;
+    if (lookIntoParentDirectories) {
+        // PIPENV_MAX_DEPTH tells pipenv the maximum number of directories to recursively search for
+        // a Pipfile, defaults to 3: https://pipenv.pypa.io/en/latest/advanced/#pipenv.environments.PIPENV_MAX_DEPTH
+        const maxDepth = getEnvironmentVariable('PIPENV_MAX_DEPTH');
+        if (maxDepth) {
+            depthToSearch = +maxDepth;

```

Comment:
Is this int coercion?  My understanding is that the safe way to do this is:

```suggestion
        const maxDepthStr = getEnvironmentVariable('PIPENV_MAX_DEPTH');
        if (maxDepthStr) {
            const maxDepth = parseInt(maxDepthStr, 10);
            if (maxDepth === undefined) {
                ...
            } else {
                depthToSearch = maxDepth;
            }
```

Reply:
Good catch. Although, `maxDepth` cannot be `undefined` ([more](https://stackoverflow.com/a/14668510/8152786)). But I see places where it can be `NaN`. Hence I'm using `isNaN(maxDepth)` method instead.